src/collectives.cc

```diff
@@ -98,15 +98,19 @@ ncclResult_t ncclAllGather_impl(const void* sendbuff, void* recvbuff, size_t sen
   NvtxParamsAllGather payload{sendcount * ncclTypeSize(datatype), datatype};
   NVTX3_FUNC_WITH_PARAMS(AllGather, AllGatherSchema, payload)
 
+  struct ncclInfo info = { ncclFuncAllGather, "AllGather",
+    sendbuff, recvbuff, sendcount, datatype, ncclSum, 0, comm, stream, /* Args */
+    ALLGATHER_CHUNKSTEPS, ALLGATHER_SLICESTEPS };
+
+  if (!mscclIsCaller()) // when msccl falls back to
+    NCCLCHECK(Recorder::instance().record(info, rrAllGather));
```

Comment:
Please make this more consistent with the other record() function calls with respect to arguments.  record(rrAllGather, info);